tests/rptest/tests/group_membership_test.py

```diff
@@ -113,3 +116,187 @@ def _check_no_duplicates():
+
+        super(GroupMetricsTest, self).__init__(test_context=ctx,
+                                               num_brokers=3,
+                                               extra_rp_conf=extra_rp_conf)
+        self._ctx = ctx
+        random.seed(42)
+
+    def _wait_topic_is_ready(self, topic):
+        # Wait for the partition to have a leader (`rpk produce` errors
+        # out if it tries to write data before this)
+        def partition_ready():
+            return KafkaCat(self.redpanda).get_partition_leader(
+                topic, 0)[0] is not None
+
+        wait_until(partition_ready, timeout_sec=10, backoff_sec=0.5)
+
+    def _get_offset_from_metrics(self, group):
+        metric = self.redpanda.metrics_sample("kafka_group_offset")
+        if metric is None:
+            return None
+        metric = metric.label_filter(dict(group=group))
+        res = {}
+        for sample in metric.samples:
+            res[(sample.labels['topic'],
+                 sample.labels['partition'])] = sample.value
+        return res
+
+    @cluster(num_nodes=3)
+    def test_check_value(self):
+        rpk = RpkTool(self.redpanda)
+
+        topic = "test_topic"
+        spec = TopicSpec(name=topic, partition_count=1)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic)
+
+        for i in range(100):
+            payload = str(random.randint(0, 1000))
+            offset = rpk.produce(topic, "", payload, timeout=5)
+
+        group_1 = "g1"
+        metric_key = (topic, "0")
+        for i in range(10):
+            rpk.consume(topic, group=group_1, n=10)
+            metrics_offsets = self._get_offset_from_metrics(group_1)
+            assert metric_key in metrics_offsets
+            assert metrics_offsets[metric_key] == (i + 1) * 10
+
+        group_2 = "g2"
+        rpk.consume(topic, group=group_2, n=50)
+        gr_2_metrics_offsets = self._get_offset_from_metrics(group_2)
+        assert metric_key in gr_2_metrics_offsets
+        assert gr_2_metrics_offsets[metric_key] == 50
+
+        rpk.group_seek(group_1, to_group=group_2)
+        gr_1_metrics_offsets = self._get_offset_from_metrics(group_1)
+        assert metric_key in gr_1_metrics_offsets
+        assert gr_1_metrics_offsets[metric_key] == 50
+
+        rpk.group_seek(group_2, to="start")
+        gr_2_metrics_offsets = self._get_offset_from_metrics(group_2)
+        assert metric_key in gr_2_metrics_offsets
+        assert gr_2_metrics_offsets[metric_key] == 0
+
+        self.redpanda.delete_topic(topic)
+        metrics_offsets = self._get_offset_from_metrics(group_1)
+        assert metrics_offsets is None
+
+    @cluster(num_nodes=3)
+    def test_multiple_topics_and_partitions(self):
+        rpk = RpkTool(self.redpanda)
+
+        topic_1 = "test_topic_1"
+        spec = TopicSpec(name=topic_1, partition_count=3, replication_factor=3)
+        self.redpanda.create_topic(spec)
+        self._wait_topic_is_ready(topic_1)
```

Comment:
arrange for all the topics to be auto-created by the test harness (see comment above)